sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2924,6 +2924,18 @@ object SQLConf {
       .stringConf
       .createWithDefault("csv,json,orc,parquet")
 
+  val DEFAULT_COLUMN_JSON_GENERATOR_FORCE_NULL_FIELDS =
+    buildConf("spark.sql.defaultColumn.jsonGeneratorForceNullFields")
```

Comment:
Since we have a conf `spark.sql.jsonGenerator.ignoreNullFields` already, I suggest making this one a JSON data source configuration as well. How about `spark.sql.jsonGenerator.writeNullIfWithDefaultValue`?

Reply:
Sounds good, done!